---
title: "Sc analysis"
author: "Luisa Santus"
date: "1/22/2020"
output: html_document
---


# Imports 


```{r include=FALSE}
library(Seurat); library(ggplot2)
# Define paths for data
source("/home/bscuser/Escritorio/cluster/02.Ebola_sc/projects_local/utils/00.datapath.R")

# ---------------------------------------------------
#           Load  references 
# ---------------------------------------------------
# Leave commented for now
# immune.combined_exvivo <- readRDS(file.path(data_path,"/02_scRNA-Seq_PBMCs/00_scRNA-Seq_exVivo_rhemac10/05_RObjects/03_prep/03_immune.combined.ready_COPY.rds"))
# 0 . Add infection status
# ebola_genome_percentage_df_exvivo <- readRDS(file.path(data_path, "02_scRNA-Seq_PBMCs/00_scRNA-Seq_exVivo_rhemac10/05_RObjects/03_prep/df_viralpercentage.rds"))
# table(colnames(immune.combined_exvivo) == rownames(ebola_genome_percentage_df_exvivo))
# immune.combined_exvivo$viral_load <- ebola_genome_percentage_df_exvivo$percentage_viral_reads
# immune.combined_exvivo$infection <- ebola_genome_percentage_df_exvivo$classification
#saveRDS(immune.combined_exvivo, file.path(data_path,"/02_scRNA-Seq_PBMCs/00_scRNA-Seq_exVivo_rhemac10/05_RObjects/03_prep/03_immune.combined.ready_infection_COPY.rds"))

immune.combined_exvivo <- readRDS(file.path(data_path,"/02_scRNA-Seq_PBMCs/00_scRNA-Seq_exVivo_rhemac10/05_RObjects/03_prep/03_immune.combined.ready_infection_COPY.rds"))


# Example to select cells with viral load higher than 0 
immune.combined_exvivo_viral_load <- immune.combined_exvivo[,immune.combined_exvivo$viral_load > 0]

```

# ----------------------------------
#     EXVIVO ANALYSIS
# ----------------------------------

```{r}
#-------- 1. Cell transcripts vs viral load 
#select late stage
immune.combined_exvivo_late_stage <- immune.combined_exvivo_viral_load[,immune.combined_exvivo_viral_load$dpi == "H024"]
immune.combined_exvivo_late_stage <- immune.combined_exvivo_late_stage[,immune.combined_exvivo_late_stage$cond == "live"]

#create df with viral load and Number of transcripts
viral_load <- as.data.frame(immune.combined_exvivo_late_stage$viral_load)
nCount_RNA <- as.data.frame(immune.combined_exvivo_late_stage$nCount_RNA)
status <- as.data.frame(immune.combined_exvivo_late_stage$infection)
late_stage_all_cells <-cbind(viral_load, nCount_RNA, status)
colnames(late_stage_all_cells) <- c("viral_load", "Ntranscripts", "status")
levels(late_stage_all_cells$status) <- c("Infected", "Not infected")

#dotplot
ggplot(late_stage_all_cells, aes(x=Ntranscripts, y=viral_load, color=status)) + geom_point()+ scale_y_log10() + scale_x_log10()+ ylab("Viral Load") + xlab("# Transcripts")

```

```{r}
#-------- 2. Viral load vs number of cells (late stage)

#all cells-all stages
viral_load <- as.data.frame(immune.combined_exvivo_viral_load$viral_load)
nCount_RNA <- as.data.frame(immune.combined_exvivo_viral_load$nCount_RNA)
status <- as.data.frame(immune.combined_exvivo_viral_load$infection)
all_all_cells <-cbind(viral_load, nCount_RNA, status)
colnames(all_all_cells) <- c("viral_load", "Ntranscripts", "status")
levels(late_stage_all_cells$status) <- c("Infected", "Not infected")

max(all_all_cells$viral_load)
ggplot(all_all_cells, aes(x=viral_load)) + geom_histogram(fill="#2E86C1", binwidth = 0.03)+ylab("# Cells")+xlab("Viral Load")+scale_x_log10()
  #xlim(c(0,50))+ylim(c(0,15))

#all cells - late
ggplot(late_stage_all_cells, aes(x=viral_load)) + geom_histogram(fill="#2E86C1", binwidth = 0.03)+ylab("# Cells")+xlab("Viral Load")+scale_x_log10()


#infected cells 
late_stage_all_cells_inf <- late_stage_all_cells[late_stage_all_cells$status == "Infected",]
ggplot(late_stage_all_cells_inf, aes(x=viral_load)) + geom_histogram(fill="#2E86C1", binwidth = 0.03)+ylab("# Cells")+xlab("Viral Load")+scale_x_log10()

length(late_stage_all_cells_inf$viral_load) == length(late_stage_all_cells$viral_load)

min(late_stage_all_cells_inf$viral_load)
max(late_stage_all_cells_inf$viral_load)
```
```{r}
#-------- 3. Number of transcripts vs number of cells 

## ALL CELLS ALL STAGES 
ggplot(late_stage_all_cells, aes(x=Ntranscripts)) + geom_histogram(fill="#2E86C1", binwidth = 0.007)+ylab("# Cells")+xlab("# Transcripts")

#select late stage
immune.combined_exvivo_viral_load

#create df with viral load and Number of transcripts
viral_load_all_cells <- as.data.frame(immune.combined_exvivo_viral_load$viral_load)
nCount_RNA_all_cells <- as.data.frame(immune.combined_exvivo_viral_load$nCount_RNA)
status_all_cells<- as.data.frame(immune.combined_exvivo_viral_load$infection)

all_stages_all_cells <-cbind(viral_load_all_cells, nCount_RNA_all_cells, status_all_cells)
colnames(all_stages_all_cells) <- c("viral_load", "Ntranscripts", "status")
levels(all_stages_all_cells$status) <- c("Infected", "Not infected")

ggplot(all_stages_all_cells, aes(x=Ntranscripts)) + geom_histogram(fill="#2E86C1", binwidth = 0.007)+ylab("# Cells")+xlab("# Transcripts")+scale_x_log10()


## LATE STAGE 
#all cells
ggplot(late_stage_all_cells, aes(x=Ntranscripts)) + geom_histogram(fill="#2E86C1", binwidth = 0.007)+ylab("# Cells")+xlab("# Transcripts")+scale_x_log10()

#infected cells 
late_stage_all_cells_inf <- late_stage_all_cells[late_stage_all_cells$status == "Infected",]
ggplot(late_stage_all_cells_inf, aes(x=Ntranscripts)) + geom_histogram(fill="#2E86C1", binwidth = 0.007)+ scale_x_log10()+ylab("# Cells")+xlab("# Transcripts")


```


## Upper UMI threshold 


```{r}
# ---------------------------------------------------
#           Load  references 
# ---------------------------------------------------
# Leave commented for now
# immune.combined_exvivo <- readRDS(file.path(data_path,"/02_scRNA-Seq_PBMCs/00_scRNA-Seq_exVivo_rhemac10/05_RObjects/03_prep/03_immune.combined.ready_COPY.rds"))
# 0 . Add infection status
# ebola_genome_percentage_df_exvivo <- readRDS(file.path(data_path, "02_scRNA-Seq_PBMCs/00_scRNA-Seq_exVivo_rhemac10/05_RObjects/03_prep/df_viralpercentage.rds"))
# table(colnames(immune.combined_exvivo) == rownames(ebola_genome_percentage_df_exvivo))
# immune.combined_exvivo$viral_load <- ebola_genome_percentage_df_exvivo$percentage_viral_reads
# immune.combined_exvivo$infection <- ebola_genome_percentage_df_exvivo$classification
#saveRDS(immune.combined_exvivo, file.path(data_path,"/02_scRNA-Seq_PBMCs/00_scRNA-Seq_exVivo_rhemac10/05_RObjects/03_prep/03_immune.combined.ready_infection_COPY.rds"))

immune.combined_exvivo.umi <- readRDS(file.path(data_path,"02_scRNA-Seq_PBMCs/00_scRNA-Seq_exVivo_rhemac10_NOVELFILTERTEST/05_RObjects/01_QC/immune.combined_qc.rds"))

# 0 . Add infection status
expression_matrix <- immune.combined_exvivo.umi@assays$RNA@counts
colnames(expression_matrix) <- Idents(immune.combined_exvivo.umi)

ebola_genes <- readRDS(file.path(data_path,"/02_scRNA-Seq_PBMCs/00_scRNA-Seq_exVivo_rhemac10/05_RObjects/05_stats/ebola_genes.rds"))

# Per cell, check the % of ebola reads ( not normalized - just calculate the % of raw counts )
ebola_reads <- expression_matrix[rownames(expression_matrix) %in% ebola_genes, ]
ebola_genome_reads <-colSums(ebola_reads)


total_reads <- colSums(expression_matrix)
ebola_gene_percentage <- (ebola_reads/total_reads)*100
ebola_genome_percentage<- (ebola_genome_reads/total_reads)*100

ebola_genome_percentage_df <- data.frame(percentage_viral_reads = ebola_genome_percentage,
                                         celltype = names(ebola_genome_percentage))

table(colnames(immune.combined_exvivo.umi) == rownames(ebola_genome_percentage_df))
immune.combined_exvivo.umi$viral_load <- ebola_genome_percentage_df$percentage_viral_reads
#immune.combined_exvivo.umi$infection <- ebola_genome_percentage_df$classification
#saveRDS(immune.combined_exvivo, file.path(data_path,"/02_scRNA-Seq_PBMCs/00_scRNA-Seq_exVivo_rhemac10/05_RObjects/03_prep/03_immune.combined.ready_infection_COPY.rds"))

# Example to select cells with viral load higher than 0 
immune.combined_exvivo.umi_viral_load <- immune.combined_exvivo.umi[,immune.combined_exvivo.umi$viral_load > 0]

table(immune.combined_exvivo.umi$viral_load > 0)
```



```{r}

#select the different cells between the two objects --> cells that we are not taking into account for further analysis 
length(colnames(immune.combined_exvivo.umi)) #61033 cells
length(colnames(immune.combined_exvivo)) #48350 cells
cells_excluded <- immune.combined_exvivo.umi[,!colnames(immune.combined_exvivo.umi) %in% colnames(immune.combined_exvivo)]
length(colnames(cells_excluded)) #12683

#get the dataframe with viral load and number of transcripts
viral_load.excluded <- as.data.frame(cells_excluded$viral_load)
nCount_RNA.excluded <- as.data.frame(cells_excluded$nCount_RNA)
cells_excluded.df <-cbind(viral_load.excluded, nCount_RNA.excluded)
colnames(cells_excluded.df) <- c("viral_load", "Ntranscripts")
cells_excluded.df <- cells_excluded.df[cells_excluded.df$viral_load > 0,]
length(cells_excluded.df$viral_load) #2411
max(cells_excluded.df$viral_load)


#-------- 1. Cell transcripts vs viral load 
ggplot(cells_excluded.df, aes(x=Ntranscripts, y=viral_load)) + geom_point()+ scale_y_log10() + scale_x_log10()+ ylab("Viral Load") + xlab("# Transcripts")

order(cells_excluded.df$viral_load)
max(cells_excluded$viral_load)

table(cells_excluded$viral_load > 20)

#-------- 2. Viral load vs number of cells (late stage)

ggplot(cells_excluded.df, aes(x=viral_load)) + geom_histogram(fill="#2E86C1", binwidth = 0.05)+ylab("# Cells")+xlab("Viral Load")+scale_x_log10()

#-------- 3. Number of transcripts vs number of cells 

ggplot(cells_excluded.df, aes(x=Ntranscripts)) + geom_histogram(fill="#2E86C1") +ylab("# Cells")+xlab("# Transcripts")+scale_x_log10()+xlim(c(0, 10000))
max(cells_excluded.df$Ntranscripts)


```




```{r}
#-------- 1. Cell transcripts vs viral load 

#select late stage
#immune.combined_exvivo.umi_late_stage <- immune.combined_exvivo.umi_viral_load[,immune.combined_exvivo.umi_viral_load$dpi == "H024"]
#immune.combined_exvivo.umi_late_stage <- immune.combined_exvivo.umi_late_stage[,immune.combined_exvivo.umi_late_stage$cond == "live"]

#immune.combined_exvivo.umi <- immune.combined_exvivo.umi[immune.combined_exvivo.umi$viral_load,]

#create df with viral load and Number of transcripts
viral_load.umi <- as.data.frame(immune.combined_exvivo.umi$viral_load)
nCount_RNA.umi <- as.data.frame(immune.combined_exvivo.umi$nCount_RNA)
#status <- as.data.frame(immune.combined_exvivo.umi_late_stage$infection)
late_stage_all_cells.umi <-cbind(viral_load.umi, nCount_RNA.umi)
colnames(late_stage_all_cells.umi) <- c("viral_load", "Ntranscripts")
#levels(late_stage_all_cells$status) <- c("Infected", "Not infected")

#dotplot
ggplot(late_stage_all_cells.umi, aes(x=Ntranscripts, y=viral_load)) + geom_point()+ scale_y_log10() + scale_x_log10()+ ylab("Viral Load") + xlab("# Transcripts")

```

```{r}
#-------- 2. Viral load vs number of cells (late stage)

#all cells
ggplot(late_stage_all_cells.umi, aes(x=viral_load)) + geom_histogram(fill="#2E86C1", binwidth = 0.05) + scale_x_log10()+ylab("# Cells")+xlab("Viral Load")


#infected cells 
late_stage_all_cells_inf <- late_stage_all_cells[late_stage_all_cells$status == "Infected",]
ggplot(late_stage_all_cells_inf, aes(x=viral_load)) + geom_histogram(fill="#2E86C1", binwidth = 0.015) + scale_x_log10()+ylab("# Cells")+xlab("Viral Load")


```
```{r}
#-------- 3. Number of transcripts vs number of cells 

## ALL CELLS ALL STAGES 
ggplot(late_stage_all_cells.umi, aes(x=Ntranscripts)) + geom_histogram(fill="#2E86C1", binwidth = 0.03)+ scale_y_log10() + scale_x_log10()+ylab("# Cells")+xlab("# Transcripts")

#select late stage
immune.combined_exvivo_viral_load <- Å‚


#create df with viral load and Number of transcripts
viral_load_all_cells <- as.data.frame(immune.combined_exvivo_viral_load$viral_load)
nCount_RNA_all_cells <- as.data.frame(immune.combined_exvivo_viral_load$nCount_RNA)
status_all_cells<- as.data.frame(immune.combined_exvivo_viral_load$infection)

all_stages_all_cells <-cbind(viral_load_all_cells, nCount_RNA_all_cells, status_all_cells)
colnames(all_stages_all_cells) <- c("viral_load", "Ntranscripts", "status")
levels(all_stages_all_cells$status) <- c("Infected", "Not infected")

ggplot(all_stages_all_cells, aes(x=Ntranscripts)) + geom_histogram(fill="#2E86C1", binwidth = 0.007)+ scale_y_log10() + scale_x_log10()+ylab("# Cells")+xlab("# Transcripts")


## LATE STAGE 
#all cells
ggplot(late_stage_all_cells, aes(x=Ntranscripts)) + geom_histogram(fill="#2E86C1", binwidth = 0.007)+ scale_y_log10() + scale_x_log10()+ylab("# Cells")+xlab("# Transcripts")

#infected cells 
late_stage_all_cells_inf.umi <- late_stage_all_cells.umi[late_stage_all_cells.umi$status == "Infected",]
ggplot(late_stage_all_cells_inf.umi, aes(x=Ntranscripts)) + geom_histogram(fill="#2E86C1", binwidth = 0.007)+ scale_y_log10() + scale_x_log10()+ylab("# Cells")+xlab("# Transcripts")

```



# Viral load percentage
```{r}
viral_load <- as.data.frame(immune.combined_exvivo$viral_load)
colnames(viral_load) <- c("viral_load")

viral_load.umi <- as.data.frame(immune.combined_exvivo.umi$viral_load)
colnames(viral_load.umi) <- c("viral_load")

viral_load.discarted <- as.data.frame(cells_excluded$viral_load)
colnames(viral_load.discarted) <- c("viral_load")

get_viral_load <- function(vl){
vl$per <- 0
vl[vl$viral_load > 1,]$per <- "1-10"
vl[vl$viral_load > 10,]$per <- "10-20"
vl[vl$viral_load > 20,]$per <- "20-30"
vl[vl$viral_load > 30,]$per <- "30-40"
vl[vl$viral_load > 40,]$per <- "40-50"
levels(vl$per) <- c("1-10", "10-20", "20-30", "30-40")
vl <- vl[vl$per != 0,]
return(vl)
}

viral_load.p <- get_viral_load(viral_load)
viral_load.p$type <- "used"
      

viral_load.discarted.p <- get_viral_load(viral_load.discarted)
viral_load.discarted.p$type <- "discarted"
table(viral_load.discarted.p$per)

all <- rbind(viral_load.p, viral_load.discarted.p)


ggplot(all, aes(x=type, fill=as.factor(per))) + geom_bar()+scale_y_continuous(breaks=seq(0, 700, by=100))



table(all$type, all$per)


```





